File: tests/python/sample.py
class Animal:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    def get_name(self):
        return self.name

    def get_age(self):
        return self.age

    @staticmethod
    def of(name, age):
        return Animal(name, age)


class Dog(Animal):
    def __init__(self, name, age, breed):
        super().__init__(name, age)
        self.breed = breed

    @staticmethod
    def of(name, age, breed):
        return Dog(name, age, breed)

    def get_breed(self):
        return self.breed

    def bark(self):
        return f"{self.name} says: Woof!"

File: tests/java/Sample.java
public class Animal {

    private String name;
    private int age;

    public Animal(String name, int age) {
        this.name = name;
        this.age = age;
    }

    public String getName() {
        return name;
    }

    public int getAge() {
        return age;
    }

    public static Animal of(String name, int age) {
        return new Animal(name, age);
    }

    public static class Dog extends Animal {

        private String breed;

        public Dog(String name, int age, String breed) {
            super(name, age);
            this.breed = breed;
        }

        public static Dog of(String name, int age, String breed) {
            return new Dog(name, age, breed);
        }

        public String getBreed() {
            return breed;
        }
    }
}
